environments/kubernetes/kube-thanos.libsonnet

```diff
@@ -17,6 +19,30 @@ local rolebinding = k.rbac.v1.roleBinding;
 (import 'thanos-receive-controller/thanos-receive-controller.libsonnet') +
 (import '../../components/thanos-querier-cache.libsonnet') +
 {
+  local jaegerAgent =
+    container.new('jaeger-agent', 'jaegertracing/jaeger-agent:1.14.0') +
+    container.withArgs([
+      '--reporter.grpc.host-port=dns:///jaeger-collector-headless.%s.svc:14250' % $.thanos.namespace,
+      '--reporter.type=grpc',
+      '--jaeger.tags=pod.namespace=${NAMESPACE},pod.name=${POD}',
+    ]) +
+    container.withEnv([
+      containerEnv.fromFieldPath('NAMESPACE', 'metadata.namespace'),
+      containerEnv.fromFieldPath('POD', 'metadata.name'),
+    ]) +
+    container.withPorts([
+      container.portsType.newNamed(6831, 'jaeger-thrift'),
+      container.portsType.newNamed(5778, 'configs'),
+    ]),
+
+  local jaegerAgentFlag = |||
+    --tracing.config=
+      type: JAEGER
+      config:
+        agent_host: 'localhost'
```

Comment:
If it's using the default Jaeger Go Client, you don't need any config at all, as it defaults to this exact setting. You might want to set the sampling strategy explicitly though, as the default is a probabilistic sampler with a 1 in 1000 chance (0.001).

Reply:
I believe that I need to provide "something" so that Thanos will enable tracing as it's turned off otherwise. But good to know, then I'll rather adjust the sampling to get everything for now.